lib/moon/design/drawer.ex

```diff
@@ -0,0 +1,32 @@
+defmodule Moon.Design.Drawer do
+  @moduledoc false
+
+  use Moon.StatelessComponent
```

Comment:
imho the component should be stateful, so than we can put default on_close event. that makes sense bc in most cases users will want to just to close drawer on-click-away